src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -438,11 +438,20 @@ internal IntPtr FreeGCHandle(IntPtr objHandle)
             return FreeGCHandle(new QCallTypeHandle(ref nativeHandle), objHandle);
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern int GetNumVirtuals(RuntimeType type);
```

Comment:
This change adds used observable overhead to reflection APIs. For example, the following ad-hoc perf test is 10% slower on my machine (~1130ms per iteration before this change, ~1260ms per iteration after this change):

```csharp
using System;
using System.Diagnostics;

var mi = typeof(A).GetMethod("M");

for (;;)
{
    Stopwatch sw = new();
    sw.Restart();
    for (int i = 0; i < 10000000; i++)
        mi.GetBaseDefinition();
    Internal.Console.WriteLine(sw.ElapsedMilliseconds.ToString());
}

class A
{
   public virtual void M() { }
}
```

Reply:
I would keep GetNumVirtuals as an FCall. just replace FCThrow with returning 0 and/or asserts. I think all calleers guaranteed that the type is != null and !IsGenericVariable. Returning 0 would just be a defensive programming.